---
editor_options: 
  chunk_output_type: console
header-includes:
   - \usepackage{setspace}
   - \singlespacing
   - \usepackage{fancyhdr}  
   - \pagestyle{fancy}      
   - \fancyhf{}             
   - \fancyfoot[C]{\thepage}
   - \renewcommand{\headrulewidth}{0pt}
output:
  bookdown::pdf_document2:
    number_sections: false
    latex_engine: xelatex
    toc: false
geometry: margin=1in
papersize: a4
pagestyle: empty
fontsize: 11 pt 
---

```{r echo=FALSE, out.height = "9%", fig.align = "left"}
    knitr::include_graphics("images/bold_cornell_logo_pms187_red.pdf")
```

July 18th, 2025

\vspace{1em}

Dear ISME Communications Editors,

We are excited to submit our manuscript, “*Interpreting UniFrac with Absolute Abundance: A Conceptual and Practical Guide*,” for consideration as a Brief Communication in *ISME Communications*.

This study introduces *Absolute UniFrac*, a conceptual and practical extension of the widely used UniFrac metric that incorporates absolute abundance information. While traditional UniFrac operates on relative abundance and overlooks variation in microbial load, our method integrates phylogenetic structure with total cell counts to reveal ecological patterns that would otherwise remain hidden.

Through simulations and a freshwater microbial case study, we demonstrate that Absolute UniFrac increases sensitivity to biologically meaningful shifts, improves statistical power, and enhances interpretability. We also clarify how microbial load variation, particularly along deep branches, can influence results and we offer guidance for thoughtful application.  We extend the Generalized UniFrac framework (Chen et al., 2012, *Bioinformatics*) by integrating absolute abundance via a tunable $\alpha$ parameter, offering a flexible, absolute abundance-aware tool for beta diversity analysis.

By bridging a foundational phylogenetic method with the realities of modern microbiome quantification, our work fills a critical methodological gap. It brings UniFrac into the next era of microbial ecology, where absolute abundance will be essential for comparing communities across space, time, and ecosystems of all kinds.

All code and data to reproduce our analyses are publicly available at [https://github.com/MarschmiLab/Pendleton_2025_Absolute_Unifrac_Paper](https://github.com/MarschmiLab/Pendleton_2025_Absolute_Unifrac_Paper).

Thank you for your consideration. We would be honored to contribute to *ISME Communications*.

Sincerely, 

Mar (Marian) Schmidt, PhD  
Assistant Professor, Department of Microbiology  
Faculty Fellow, Atkinson Center for a Sustainable Future  
Cornell University  
https://marschmilab.github.io/  