---
editor_options: 
  chunk_output_type: console
bibliography: references_updated.json
csl: the-isme-journal.csl
header-includes:
   - \usepackage{lineno}
   - \linenumbers
   - \usepackage{setspace}
   - \doublespacing
   - \usepackage{fancyhdr}  
   - \pagestyle{fancy}      
   - \fancyhf{}             
   - \fancyfoot[C]{\thepage}
   - \renewcommand{\headrulewidth}{0pt}
output:
  bookdown::pdf_document2:
    number_sections: false
    latex_engine: xelatex
    toc: false
geometry: margin=1in
papersize: a4
pagestyle: empty
fontsize: 12 pt 
---

\singlespacing


\newpage

## Interpreting UniFrac with Absolute Abundance: A Conceptual and Practical Guide

Augustus Pendleton^1^\* & Marian L. Schmidt^1^\*

^1^Department of Microbiology, Cornell University, 123 Wing Dr, Ithaca, NY 14850, USA

**Corresponding Authors:** Augustus Pendleton: [arp277\@cornell.edu](mailto:arp277@cornell.edu){.email}; Marian L. Schmidt: [marschmi\@cornell.edu](mailto:marschmi@cornell.edu){.email}

**Author Contribution Statement:** Both authors contributed equally to the manuscript.

**Preprint servers:** This article was submitted to *bioRxiv* (doi: ) under a CC-BY-NC-ND 4.0 International license.

**Keywords:** Microbial Ecology - Beta Diversity - Absolute Abundance - Bioinformatics - UniFrac

**Data Availability:** All data and code used to produce the manuscript are available at <https://github.com/MarschmiLab/Pendleton_2025_Absolute_Unifrac_Paper>, in addition to a reproducible `renv` environment. All packages used for analysis are listed in Table S1.

\newpage

    Microbial ecologists routinely compare communities using β-diversity metrics derived from relative abundances. Yet this approach overlooks a critical ecological dimension: microbial load. Across ecosystems, ranging from gut to soil to water, total microbial abundance can vary by orders of magnitude, even among healthy systems. Relative data can misrepresent true shifts in biomass, making taxa appear to change when they have not. High-throughput sequencing produces compositional data, in which each taxon’s abundance is constrained by all others [@gloor2017]. In low-biomass samples, this distortion is especially pronounced, allowing contaminants to appear biologically meaningful when absolute counts are unknown.

    Recognition of these limitations has reshaped how we interpret microbial variation. Quantitative profiling studies show that cell abundance, not only composition, can drive major community differences and alter co-occurrence networks [@vandeputte2017]. To overcome compositional constraints, researchers increasingly use flow cytometry, qPCR, and genomic spike-ins to quantify microbial load [@props2017; @wang2021]. These tools improve detection of functionally relevant taxa and mitigate the compositional constraints imposed by sequencing [@gloor2017; @vandeputte2017]. Most studies using absolute data rely on Bray-Curtis dissimilarity, which does not require normalization to proportions [e.g. @props2017; @rao2021]. Weighted UniFrac remains the dominant phylogenetic β-diversity metric, yet its behavior under absolute abundance frameworks remains unclear. Here, we present *Absolute UniFrac*, a direct extension of Weighted UniFrac that incorporates total abundance, and evaluate its impact across simulated and real-world datasets.

    The UniFrac distance was first introduced by Lozupone & Knight (2005), and has since become enormously popular as a measure of $\beta$-diversity within the field of microbial ecology [@lozupone2005]. A benefit of the UniFrac distance is that is considers phylogenetic information when estimating the distance between two communities. After first generating a phylogenetic tree representing species (or amplicon sequence variants, "ASVs") from all samples, the UniFrac distance computes the fraction of branch-lengths which is *shared* between communities, relative to the total branch length represented in the tree. UniFrac can be both unweighted, in which only the incidence of species is considered, or weighted, wherein a branch's contribution is weighted by the proportional abundance of taxa on that branch [@lozupone2007]. The weighted UniFrac is derived: $$
U^R=\frac{\sum_{i=1}^{n} b_i|p_i^a-p_i^b|}{\sum_{i=1}^{n} b_i(p_i^a+p_i^b)}
$$ where we weight the length of each branch, $b_i$, by the difference in the relative abundance of all species ($p_i$) descended from that branch in sample $a$ or sample $b$. Here, we denote this distance as *U^R^*, for "Relative Unifrac". Popular packages which calculate weighted Unifrac-including `diversity-lib` QIIME plug-in and the R packages `phyloseq` and `GUniFrac`-run this normalization by default.

    Because $U^R$ is most sensitive to changes in abundant lineages, it can sometimes obscure compositional differences driven by rare to moderately-abundant taxa [@chen2012]. To address this weakness, Chen et al. (2012) introduced the generalized UniFrac distance ($GU^R$), in which the impact of abundant lineages can be mitigated by decreasing the parameter $\alpha$: $$
GU^R=\frac{\sum\limits_{i=1}^{n} b_i(p_i^a+p_i^b)^\alpha\left|\dfrac{p_i^a-p_i^b}{p_i^a+p_i^b}\right|}{\sum\limits_{i=1}^{n} b_i(p_i^a+p_i^b)^\alpha}
$$

where $\alpha$ ranges from 0 (close to unweighted UniFrac) up to 1 (identical to $U^R$, above). However, if one wishes to use absolute abundances, both $U^R$ and $GU^R$ can be derived without normalizing to proportions:

$$
\begin{aligned}
U^A=\frac{\sum\limits_{i=1}^{n}b_i|c_i^a-c_i^b|}{\sum\limits_{i=1}^{n}b_i(c_i^a+c_i^b)}\qquad&&
GU^A=\frac{\sum\limits_{i=1}^{n}b_i(c_i^a+c_i^b)^\alpha\left|\dfrac{c_i^a-c_i^b}{c_i^a+c_i^b}\right|}{\sum\limits_{i=1}^{n}b_i(c_i^a+c_i^b)^\alpha}
\end{aligned}
$$

Where $c_i^a$ and $c_i^b$ stands for the absolute counts of species descended from branch $b_i$ in community $a$ and $b$, respectively. We refer to these distances as "Absolute Unifrac" and "Generalized Absolute Unifrac" ($U^A$ and $GU^A$) .

    To illustrate how $U^A$ behaves, we constructed a simulated community of four ASVs arranged in a simple, balanced phylogeny (Fig. 1A). By varying the absolute abundance of each ASV (1, 10, or 100), we generated 81 samples and 3,240 pairwise comparisons. For each pair, we computed four dissimilarity metrics: Bray-Curtis with relative abundance ($BC^R$), Bray-Curtis with absolute abundance ($BC^A$), Weighted UniFrac with relative abundance ($U^R$), and Weighted UniFrac with absolute abundance ($U^A$). 

    $U^A$ does not consistently yield higher or lower distances but instead varies depending on how abundance and phylogeny intersect (Fig. 1B). In the improbable scenario that all branch lengths are equal, $U^A$ is always less than or equal to $BC^A$ (Fig. S1), but real-world trees rarely behave this way. These comparisons emphasize that incorporating phylogeny and absolute abundance reshapes distance estimates in nontrivial ways.

    To better understand how these metrics diverge, we examined individual sample pairs (Fig. 1C). Scenario 1 (gold star) illustrates the classic advantage of UniFrac: ASV_1 and ASV_2 are phylogenetically close, so $U^R$ and $U^A$ discern greater similarity between samples than $BC^R$ and $BC^A$, which ignore phylogenetic structure. Scenario 2 (red star) highlights a limitation of relative metrics: two samples with identical relative composition but a 100-fold difference in biomass appear identical to $BC^R$ and $U^R$, but not to their absolute counterparts. In Scenario 3 (green star), incorporating absolute abundance decreases dissimilarity. $BC^A$ and $U^A$ are lower than their relative counterparts because half the community is identical in absolute terms, despite proportional differences. In contrast, Scenario 4 (blue star) shows that $U^A$ can increase dissimilarity relative to $BC^A$ when abundance differences occur on long branches, amplifying phylogenetic dissimilarity. 

    Across all 3,240 pairwise comparisons, $U^A$ is usually smaller than and strongly correlated with $BC^A$ (Pearsons $r$ = 0.82, $p$ \< 0.0001) than with $BC^R$ ($r$ = 0.41) and $U^R$ ($r$ = 0.55), reflecting the effect illustrated in Scenario 1. However, exceptions like Scenario 4 show that $U^A$ can also yield larger distances than $BC^A$ when abundance differences occur on long branches. These scenarios demonstrate that $U^A$ integrates ecological realism by capturing differences in both lineage identity and total biomass, offering a nuanced view of community structure grounded in biological context, while remaining sensitive to long branches.

\begingroup \centering

![](../figures/Figure_1.png){width="100%"}

\endgroup
\begingroup \fontsize{10}{12}\selectfont

*Figure 1. Simple simulations to compare the impact of absolute abundance on phylogenetic and non-phylogenetic distance measures.* (A) We created a small community represented by 4 ASVs. We explored all permutations of each ASV holding an absolute count of 1, 10, or 100 to create 81 communities, producing 3240 comparisons for which we calculated distance metrics. (B) Distributions of the difference between weighted Unifrac using absolute abundance ($U^A$), Bray-Curtis using relative abundance ($BC^R$), weighted Unifrac using relative abundance ($U^R$), and Bray-Curtis using absolute abundance ($BC^A$). (C) Scenarios (comparisons) highlighting occasions when $U^A$ is larger or lesser compared to other metrics. The stars indicate what region of the distributions in (B) this scenario would fall. The actual values for each metric are displayed beneath the scenario.

\endgroup

    We next explored the practical effect $U^A$ has on separating groups within a real dataset. We used a previously published 16S dataset from Lake Ontario, represented by 66 samples and \>7,000 ASVs, where samples clustered into three main groups defined by depth and month due to changes in both composition and abundance (Fig. S2, [@pendleton]). We sought to demonstrate how weighting by absolute abundance can affect the interpretation and statistical power to separate samples between these depth/month groups.

    We calculated the generalized absolute UniFrac ($GU^A$) across three levels of $\alpha$: 0.0 (close to unweighted UniFrac), 0.5, and 1.0 (identical to $U^A$). Fig. 2A shows PCoA ordinations of the same samples across this $\alpha$-range. As $\alpha$ increases, we see a greater recognition of similarity between Shallow September and Shallow May, reflecting their relatively higher cell counts compared to the Deep samples. Note that an increasing amount of variation is also assigned to the first axis, going from 18.3% up to 76.7%. This trend was also true for $U^R$ across multiple $\alpha$, but to a much weaker degree (Fig. S3).

    We used PERMANOVAs to quantify the variance ($R^2$) and statistical power ($F$-statistic) our three depth-month groups can explain across different distance metrics ($GU^R$, $GU^A$, $BC^R$ and $BC^A$) and different alpha values (Fig. 2B and 2C). Absolute abundance metrics consistently allow for greater $R^2$ values ($GU^A$ reaching a maximum of 75.8%) and F-statistics (1.56X higher for $GU^A$ than $GU^R$) than their relative counterparts. For UniFrac distances, this separation increases as alpha increases. As such, the incorporation of absolute abundances can be a powerful tool to emphasize differences in microbial load between groups.

    That said, we identify a major caveat; $U^A$ can become strongly correlated to the cell-counts alone compared to other distance metrics (Fig. 2D). We used a Mantel test to assess the correlation between each distance metric to the absolute difference in cell counts between each sample. Absolute abundance measures are more sensitive to differences in cell counts compared to their relative counterparts. This is intuitive, and to a degree, intentional; the utility of these metrics is to identify changes in absolute abundance even in the case of compositional similarity. However, at $\alpha$ = 1, the value of $U^A$ becomes almost entirely based on the overall abundance of a sample. We see this reflected in the ordinations as well; as $\alpha$ increases, the first Axis increasingly represents and correlates to the absolute abundance (Spearman's $\rho$ = -1.0 for $\alpha$ = 1.0 versus Spearman's $\rho$ = 0.58 for $\alpha$ = 0.0 versus ). It is possible that any 2D structure we observed in the third panel of Fig. 2A is simply the result of the horseshore effect [@morton2017].

    With this in mind, we encourage others to think critically about how heavily they want their distance metric to correspond to absolute abundance, and modulate this effect using $GU^A$ rather than $U^A$. As a simple start from these data, we recommended using an $\alpha$ value of 0.5, consistent with the original recommendation of Chen et al. (2012) but of heightened importance when using absolute abundances.

\begingroup \centering

![](../figures/Figure_2.png){width="100%"}

\endgroup
\begingroup \fontsize{10}{12}\selectfont

*Figure 2. Impact of absolute abundance on separating freshwater microbial communities* (A) Samples represent microbial communities from Lake Ontario taken at different times (May and September) and at different depths (Shallow and Deep), as originally described in [@pendleton]. We used Principal Coordinates Analysis (PCoA) to ordinate samples based on the $GU^A$ at $\alpha$ values of 0.0, 0.5, and 1.0. The proportion of variance assigned to each axis is provided in brackets. Note that the x-axis is reversed in the first panel, to provide visual symmetry across ordinations. We then used PERMANOVAs to quantify the variance ($R^2$) (B) and statistical power ($F$-statistic) (C) our three depth-month groups can explain across different distance metrics ($GU^R$, $GU^A$, $BC^R$ and $BC^A$) and different alpha values. (D) Correlations between different distance metrics and differences in over-all cell abundances as assessed by a Mantel test.

\endgroup

    The incorporation of absolute abundance allows microbial ecologists to assess more realistic, ecologically-relevant differences in microbial communities in situations for where microbial load matters. For example, the temporal development of the infant microbiome is captured in both a rise in absolute abundance in addition to compositional changes [@rao2021]; bacteriophage predation in wastewater bioreactors can be understood only when incorporating microbial load [@shapiro2010]; and the negative impact of antibiotics on the abundance of specific taxa in the swine gut was missed using relative abundance approaches [@wagner2025]. As $\beta$-diversity metrics are an essential tool for every microbial ecologist-and UniFrac distances are highly popular within microbiome research specifically-we encourage other researchers to adopt the usage of $GU^A$ in comparing their samples.

    Our data show, however, that the interpretation of these metrics can be complex, and researchers should consider (and justify) the relative importance of absolute abundance in their estimates of $\beta$-diversity. We also do not address other concerns relevant to estimating $\beta$-diversity, including the impacts of sampling effort on richness and whether rarefaction should be applied when calculating $GU^A$ [@schloss2023]. As ever, interpreting the results of these metrics critically and exploring the sensitivity of your conclusions across multiple metrics is encouraged [@kers2022]. Through this work, we hope to facilitate the adoption of $GU^A$ and its interpretation in an approachable manner.

#### References
