---
editor_options: 
  chunk_output_type: console
bibliography: references.json
csl: the-isme-journal.csl
header-includes:
   - \usepackage{lineno}
   - \linenumbers
   - \usepackage{setspace}
   - \doublespacing
   - \usepackage{fancyhdr}  
   - \pagestyle{fancy}      
   - \fancyhf{}             
   - \fancyfoot[C]{\thepage}
   - \renewcommand{\headrulewidth}{0pt}
output:
  bookdown::pdf_document2:
    number_sections: false
    latex_engine: xelatex
    toc: false
geometry: margin=1in
papersize: a4
pagestyle: empty
fontsize: 12 pt 
---

\singlespacing

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE,
                      options(scipen=999
                              )
                      )
library(phyloseq)
library(tidyverse)
```

\newpage

## Interpretation and application of absolute abundance in Weighted UniFrac distance

Augustus Pendleton^1^\* & Marian L. Schmidt^1^\*

^1^Department of Microbiology, Cornell University, 123 Wing Dr, Ithaca, NY 14850, USA

**Corresponding Authors:** Augustus Pendleton: [arp277\@cornell.edu](mailto:arp277@cornell.edu){.email}; Marian L. Schmidt: [marschmi\@cornell.edu](mailto:marschmi@cornell.edu){.email}

**Author Contribution Statement:** Both authors contributed equally to the manuscript.

**Preprint servers:** This article was submitted to *bioRxiv* (doi: ) under a CC-BY-NC-ND 4.0 International license.

**Keywords:** Microbial Ecology - Beta Diversity - Absolute Abundance - Bioinformatics

\newpage

    The UniFrac distance was first introduced by Lozupone & Knight (2005), and has since become enormously popular as a measure of $\beta$-diversity within the field of microbial ecology [@lozupone2005]. A major draw of the UniFrac distance is that is considers phylogenetic information when estimating the distance between two communities. After first generating a phylogenetic tree representing species (or amplicon sequence variants, ASVs) from all samples, the UniFrac distance computes the fraction of branch-lengths which is *shared* between communities, relative to the total branch length represented in the phylogenetic tree. UniFrac can be both *unweighted*, in which only the incidence of species is considered, or *weighted*, wherein the contribution a branch makes to the overall distance is weighted by the proportional abundance of taxa descended from that branch [@lozupone2007]. The weighted UniFrac is derived as: $$
U^R=\frac{\sum_{i=1}^{n} b_i|p_i^a-p_i^b|}{\sum_{i=1}^{n} b_i(p_i^a+p_i^b)}
$$ Where we weight the length of each branch, $b_i$, by the difference in the relative abundance of all species ($p_i$) descended from that branch in sample $a$ or sample $a$. As such, we denote this distance as *U^R^*, for "Relative Unifrac". Popular packages which calculate weighted Unifrac, including QIIME and the R packages `phyloseq` and `GUniFrac` run this normalization by default.

    Because $U^R$ is most sensitive to changes in abundant lineages, it can sometimes obscure compositional changes occurring in rare to moderately-abundant taxa [@chen2012]. To address this weakness, Chen et al. (2012) introduced the generalized UniFrac distance ($GU^R$), in which the impact of abundant lineages can be mitigated by decreasing the parameter $\alpha$: $$
GU^R=\frac{\sum\limits_{i=1}^{n} b_i(p_i^a+p_i^b)^\alpha\left|\dfrac{p_i^a-p_i^b}{p_i^a+p_i^b}\right|}{\sum\limits_{i=1}^{n} b_i(p_i^a+p_i^b)^\alpha}
$$ However, if one wishes to use absolute abundances, both $U^R$ and $GU^R$ can be derived without normalizing by total counts:

$$
\begin{aligned}
U^A=\frac{\sum\limits_{i=1}^{n}b_i|c_i^a-c_i^b|}{\sum\limits_{i=1}^{n}b_i(c_i^a+c_i^b)}\qquad&&
GU^A=\frac{\sum\limits_{i=1}^{n}b_i(c_i^a+c_i^b)^\alpha\left|\dfrac{c_i^a-c_i^b}{c_i^a+c_i^b}\right|}{\sum\limits_{i=1}^{n}b_i(c_i^a+c_i^b)^\alpha}
\end{aligned}
$$

Where $c_i^a$ and $c_i^b$ stands for the absolute counts of species descended from branch $b_i$ in community $a$ and $b$, respectively, yielding "Absolute Unifrac" ($U^A$/$GU^A$) c. There is growing recognition that microbial load matters, and methods for discerning absolute abundance like flow cytometry and qPCR are growing in popularity [@props2017; @wang2021]. However, it can be conceptually difficult to predict and interpret the impact of absolute abundance on complex metrics like UniFrac.

    As such, we began with a simulated community, represented by four ASVs with simple phylogenetic relationships (Fig. 1A). We assigned each ASV an absolute count of 1, 10, or 100, yielding 81 samples and 3240 cross-sample comparisons for which we calculated the Bray-Curtis dissimilarity and weighted Unifrac distance using both relative and absolute abundances ($BC^R$, $BC^A$, $U^R$, and $U^A$). We then calculated the difference in distance calculated by each metric compared to Absolute Unifrac; the distributions of these differences are shown in Fig. 1B.

    For all metrics, there are scenarios where $U^A$ estimates either greater or less distance (in the improbable scenario that all branch lengths are equal, $U^A$ will always be equal to or less than $BC^A$, Fig. S1). We then isolated specific comparisons to illustrate where and when $U^A$ differs from other metrics (Fig. 1C). Scenario one (gold star) indicates the canonical benefit of UniFrac distances: the phylogenetic similarity of ASV_1 and ASV_2 allow $U^R$ and $U^A$ to discern greater similarity between samples than the phylogenetically-oblivious $BC^R$/$BC^A$. Scenario two (red star) illustrates the importance of absolute abundances; $BC^R$ and $U^R$ perceive two communities as identical despite large differences in absolute abundance.

    At times, the incorporation of absolute abundance can also *decrease* the perceived dissimilarity (green star). Here, $BC^A$ and $U^A$ are both smaller than their relative counterparts, as over half of the community is perceived as identical, despite differences in their relative abundances due to proportionality. Finally, $U^A$ is highly correlated with $BC^A$ (Pearsons $r$ = 0.82, $p$ \< 0.0001) compared to $BC^R$ ($r$ = 0.41) and $U^R$ ($r$ = 0.55). However, there are scenarios in which $U^A$ is greater than $BC^A$ (blue star). Because the absolute differences are observed on longer branches relative to the rest of the tree, $U^A$ discerns greater distance than $BC^A$.

\begingroup \centering

![](../figures/Figure_1.png){width="75%"}

\endgroup
\begingroup \fontsize{10}{12}\selectfont

*Figure 1. Simple simulations to compare the impact of absolute abundance on phylogenetic and non-phylogenetic distance measures.* (A) We created a small community represented by 4 ASVs. We explored all permutations of each ASV holding an absolute count of 1, 10, or 100 to create 81 communities, producing 3240 comparisons. We calculated the Bray-Curtis dissimilarity and weighted Unifrac distance using both relative and absolute abundances ($BC^R$, $BC^A$, $U^R$, and $U^A$). (B) Distributions of the difference between weighted Unifrac using absolute abundance ($U^A$), Bray-Curtis using relative abundance ($BC^R$), weighted Unifrac using relative abundance ($U^R$), and Bray-Curtis using absolute abundance ($BC^A$). (C) Scenarios (comparisons) highlighting occasions when $U^A$ is larger or lesser compared to other metrics. The stars indicate what region of the distributions in (B) this scenario would fall. The actual values for each metric are displayed beneath the scenario.

\endgroup

    We next explored the practical effect $U^A$ has on separating groups across a real dataset. We used a previously published 16S dataset from Lake Ontario, where samples clustered into three main groups defined by depth and month due to changes in both composition and abundance (Fig. S2, [@pendleton]). To illustrate the impact of absolute abundance, we calculated the generalized absolute UniFrac ($GU^A$) across three levels of $\alpha$: 0.0 (close to unweighted UniFrac), 0.5, and 1.0 (identical to $U^A$). Fig. 2A shows PCoA ordinations of the same samples across this $\alpha$-range. As changes in abundance become more weighted ($\alpha$ increases), we see a greater recognition of similarity between Shallow September and Shallow May, reflecting their relatively higher cell counts compared to the Deep samples. Note that an increasing amount of variation is assigned to the first axis, going from 18.3% up to 76.7%. This trend was also true for $U^R$ across multiple $\alpha$, but to a much weaker degree (Fig. S3).

    We used PERMANOVAs to quantify the variance ($R^2$) and statistical power ($F$-statistic) our three depth-month groups can explain across different distance metrics ($GU^R$, $GU^A$, $BC^R$ and $BC^A$) and different alpha values (Fig. 2B and 2C). We see that absolute abundance metrics consistently allow for greater $R^2$ values and F-statistics than their relative counterparts. For UniFrac distances, this separation increases as alpha increases. As such, the incorporation of absolute abundances can be a powerful tool to emphasize differences in microbial load between groups.

    That said, we identify a major caveat; $U^A$ can become strongly correlated to the cell-counts alone compared to other distance metrics (Fig. 2D). We used a Mantel test to assess the correlation between each distance metric to the absolute difference in cell counts between each sample. Absolute abundance measures are more sensitive to differences in cell counts compared to their relative counterparts. This is intuitive, and to a degree, intentional; the utility of these metrics is to identify changes in absolute abundance even in the case of compositional similarity. However, at $\alpha$ = 1, the value of $U^A$ becomes almost entirely based on the overall abundance of a sample. We see this reflected in the ordinations as well; as $\alpha$ increases, the first Axis increasingly represents (and correlates to) the absolute abundance, and its possible that any 2D structure we observed in the third panel of Fig. 2A is simply the result of the horseshore effect [@morton2017].

    With this in mind, we encourage others to think critically about how heavily they want their distance metric to correspond to absolute abundance, and modulate this effect using $GU^A$ rather than $U^A$. As a simple start from these data, we recommended using an $\alpha$ value of 0.5, consistent with the original recommendation of Chen et al. (2012) but of heightened importance when using absolute abundances.

\begingroup \centering

![](../figures/Figure_2.png){width="100%"}

\endgroup
\begingroup \fontsize{10}{12}\selectfont

*Figure 2. Impact of absolute abundance on separating freshwater microbial communities* (A) Samples represent microbial communities from Lake Ontario taken at different times (May and September) and at different depths (Shallow and Deep), as originally described in Pendleton et al. (2025). We used Principal Coordinates Analysis (PCoA) to ordinate samples based on the $GU^A$ at $\alpha$ values of 0.0, 0.5, and 1.0. The proportion of variance assigned to each axis is provided in brackets. Note that the x-axis is reversed in the first panel, to provide visual symmetry across ordinations. We then used PERMANOVAs to quantify the variance ($R^2$) (B) and statistical power ($F$-statistic) (C) our three depth-month groups can explain across different distance metrics ($GU^R$, $GU^A$, $BC^R$ and $BC^A$) and different alpha values. (D) Correlations between different distance metrics and differences in over-all cell abundances as assessed by a Mantel test. 

    The incorporation of absolute abundance allows microbial ecologists to assess more realistic, ecologically-relevant differences in microbial communities. As $\beta$-diversity metrics are an essential tool for every microbial ecologist, and UniFrac distances are highly popular within microbiome research specifically, we encourage other researchers to adopt the usage of $GU^A$ in comparing their samples. This index could be especially relevant in instances where microbial load varies widely and is ecologically relevant, including the development of the infant gut, within bioreactors, and after antibiotic treatment [@rao2021; @shapiro2010; @wagner2025].

    As our data show, however, the interpretation of these metrics can be complex, as they integrate differences in composition, phylogeny, *and* abundance in ways which can be unpredictable. As ever, interpreting the results of these metrics critically, and exploring the sensitivity of your conclusions across multiple metrics, is encouraged [@kers2022].

#### References
