# Interpreting UniFrac with Absolute Abundance: A Conceptual and Practical Guide

**Authors:** Augustus Pendleton<sup>1</sup> & Marian L. Schmidt<sup>1</sup>

<sup>1</sup>Department of Microbiology, Cornell University, 123 Wing Dr, Ithaca, NY 14850, USA

## ISME Communications

**DOI Link to Article:** [https://doi.org/10.1093/ismeco/ycaf250](https://doi.org/10.1093/ismeco/ycaf250)

## bioRxiv

**DOI Link to bioRxiv:** [https://doi.org/10.1101/2025.07.18.665540](https://doi.org/10.1101/2025.07.18.665540)


## Abstract 

ğ›½-diversity is central to microbial ecology, yet commonly used metrics overlook changes in microbial load (or â€œabsolute abundanceâ€), limiting their ability to detect ecologically meaningful shifts. Popular for incorporating phylogenetic relationships, UniFrac distances currently default to relative abundance and therefore omit important variation in microbial abundances. As quantifying absolute abundance becomes more accessible, integrating this information into ğ›½diversity analyses is essential. Here, we introduce Absolute UniFrac (ğ‘ˆğ´), a variant of Weighted UniFrac that incorporates absolute abundances. Using simulations and a reanalysis of four 16S rRNA metabarcoding datasets (from a nuclear reactor cooling tank, the mouse gut, a freshwater lake, and the peanut rhizospere), we demonstrate that Absolute UniFrac captures microbial load, composition, and phylogenetic relationships. While this can improve statistical power to detect ecological shifts, we also find Absolute Unifrac can be strongly correlated to differences in cell abundances alone. To balance these effects, we also incorporate absolute abundance into the generalized extension (ğºğ‘ˆğ´) that has a tunable, continuous ecological parameter (ğ›¼) that modulates the relative contribution of rare versus abundant lineages to ğ›½-diversity calculations. Finally, we benchmark GUA and show that although computationally slower than conventional alternatives, GUA is comparably sensitive to noise in load estimates compared to conventional alternatives like Bray-Curtis dissimilarities, particularly at lower ğ›¼. By coupling phylogeny, composition, and microbial load, Absolute Unifrac integrates three dimensions of ecological change, better equipping microbial ecologists to quantitatively compare microbial communities.


