<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Augustus Pendleton">
<meta name="dcterms.date" content="2025-09-04">

<title>Making Our Phylogenetic Tree</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="02_Barlow_MakeTree_files/libs/clipboard/clipboard.min.js"></script>
<script src="02_Barlow_MakeTree_files/libs/quarto-html/quarto.js"></script>
<script src="02_Barlow_MakeTree_files/libs/quarto-html/popper.min.js"></script>
<script src="02_Barlow_MakeTree_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="02_Barlow_MakeTree_files/libs/quarto-html/anchor.min.js"></script>
<link href="02_Barlow_MakeTree_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="02_Barlow_MakeTree_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="02_Barlow_MakeTree_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="02_Barlow_MakeTree_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="02_Barlow_MakeTree_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Making Our Phylogenetic Tree</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Augustus Pendleton </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 4, 2025</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-packages-and-functions" class="level1">
<h1>Load Packages and Functions</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(phyloseq, ggtree, patchwork, tidytree, phytools, tidyverse, <span class="at">install =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data" class="level1">
<h1>Load data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in the raw_phyloseq data from DADA2_workflow.Rmd</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"clean_barlow_phylo.RData"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>clean_barlow_phylo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 885 taxa and 93 samples ]
sample_data() Sample Data:       [ 93 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 885 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
</section>
<section id="write-asv-fasta-file" class="level1">
<h1>Write ASV Fasta file</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># first pull taxonomic table from phyloseq object</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># contains ASV and sequences</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>tax_table <span class="ot">&lt;-</span> clean_barlow_phylo<span class="sc">@</span>tax_table <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ASV, ASVseqs)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add &gt; symbol to ASV names for file formatting</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>ASV <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>,tax_table<span class="sc">$</span>ASV)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># bind ASVs and sequences together in one long character</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>seqs_fasta <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rbind</span>(tax_table<span class="sc">$</span>ASV, tax_table<span class="sc">$</span>ASVseqs))</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># write seqs_fasta out as a fasta file</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(seqs_fasta, <span class="at">file =</span> <span class="st">"barlow_asvs.fasta"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mafft-for-alignment" class="level1">
<h1>MAFFT for Alignment</h1>
<p>This step aligns the ASV fasta from phyloseq_preprocessing and outputs an aligned fasta file. Note that you should be running this from the project’s home directory!</p>
<div class="cell" data-engine.opts="-l">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Run MAFFT on file</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I am using auto options for now (instead of specifying arguments)</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># MAFFT automatically recognizes that it is a nucleotide alignment</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Set seed (not sure if relevant but not taking chances)</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="va">RANDOM</span><span class="op">=</span>31491</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="ex">/programs/mafft/bin/mafft</span> <span class="at">--auto</span> <span class="at">--quiet</span> barlow_asvs.fasta <span class="op">&gt;</span> aligned_asvs.fasta </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fasttree" class="level1">
<h1>FastTree</h1>
<p>This section of code takes the aligned fasta file and runs it through Fast Tree to output an unrooted tree. It also takes awhile. The end of the logfile will tell you relevant information on who the tree was built.</p>
<div class="cell" data-engine.opts="-l">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#add to path</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span>/programs/FastTree-2.1.11:<span class="va">$PATH</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="va">RANDOM</span><span class="op">=</span>31491</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="ex">FastTree</span> <span class="at">-gtr</span> <span class="at">-nt</span> <span class="at">-fastest</span> <span class="at">-quiet</span> <span class="at">-log</span> fasttree.log aligned_asvs.fasta  <span class="op">&gt;</span> unrooted.tree</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-with-ggtree" class="level1">
<h1>Plot with ggtree</h1>
<p>First load tree and add to phyloseq object</p>
<div class="cell" data-fig.dim="[12,6]">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read tree file</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">"unrooted.tree"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rooting-our-tree" class="level1">
<h1>Rooting our tree</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">is.rooted</span>(tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Midpoint root</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>root_tree <span class="ot">&lt;-</span> <span class="fu">midpoint_root</span>(tree)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">is.rooted</span>(root_tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with our physeq</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>root_physeq <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(clean_barlow_phylo, root_tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(root_physeq) <span class="sc">+</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in psmelt(model): The sample variables: 
Sample
 have been renamed to: 
sample_Sample
to avoid conflicts with special phyloseq plot attribute names.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'tidytree'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="02_Barlow_MakeTree_files/figure-html/first-tree-view-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Okay, our tree looks good, There aren’t any crazy long branches that really worry me, or are unannotated.</p>
</section>
<section id="saving-our-final-tree" class="level1">
<h1>Saving our final tree</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(root_physeq, <span class="at">file =</span> <span class="st">"root_physeq.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="session-information" class="level1">
<h1>Session Information</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reproducibility</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.3.3 (2024-02-29)
 os       Rocky Linux 9.5 (Blue Onyx)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2025-09-04
 pandoc   3.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/tools/ (via rmarkdown)
 quarto   1.3.450 @ /usr/lib/rstudio-server/bin/quarto/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────
 ! package           * version    date (UTC) lib source
   ade4                1.7-23     2025-02-14 [1] CRAN (R 4.3.3)
   ape               * 5.8-1      2024-12-16 [1] CRAN (R 4.3.3)
 P aplot               0.2.8      2025-07-02 [?] CRAN (R 4.3.3)
   Biobase             2.68.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   BiocGenerics        0.54.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   biomformat          1.30.0     2023-10-24 [1] Bioconductor
   Biostrings          2.76.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   cachem              1.1.0      2024-05-16 [1] CRAN (R 4.3.3)
   cli                 3.6.5      2025-04-23 [1] CRAN (R 4.3.3)
   cluster             2.1.6      2023-12-01 [2] CRAN (R 4.3.3)
 P clusterGeneration   1.3.8      2023-08-16 [?] CRAN (R 4.3.2)
 P coda                0.19-4.1   2024-01-31 [?] CRAN (R 4.3.2)
   codetools           0.2-19     2023-02-01 [2] CRAN (R 4.3.3)
 P combinat            0.0-8      2012-10-29 [?] CRAN (R 4.3.2)
   crayon              1.5.3      2024-06-20 [1] CRAN (R 4.3.3)
   data.table          1.17.8     2025-07-10 [1] CRAN (R 4.3.3)
 P DEoptim             2.2-8      2022-11-11 [?] CRAN (R 4.3.2)
 P devtools            2.4.5      2022-10-11 [?] CRAN (R 4.3.2)
   digest              0.6.37     2024-08-19 [1] CRAN (R 4.3.3)
 P doParallel          1.0.17     2022-02-07 [?] CRAN (R 4.3.2)
   dplyr             * 1.1.4      2023-11-17 [1] CRAN (R 4.3.3)
 P ellipsis            0.3.2      2021-04-29 [?] CRAN (R 4.3.2)
   evaluate            1.0.5      2025-08-27 [1] CRAN (R 4.3.3)
 P expm                1.0-0      2024-08-19 [?] CRAN (R 4.3.2)
   farver              2.1.2      2024-05-13 [1] CRAN (R 4.3.3)
   fastmap             1.2.0      2024-05-15 [1] CRAN (R 4.3.3)
 P fastmatch           1.1-6      2024-12-23 [?] CRAN (R 4.3.2)
   forcats           * 1.0.0      2023-01-29 [1] CRAN (R 4.3.3)
   foreach             1.5.2      2022-02-02 [1] CRAN (R 4.3.3)
   fs                  1.6.6      2025-04-12 [1] CRAN (R 4.3.3)
   generics            0.1.4      2025-05-09 [1] CRAN (R 4.3.3)
   GenomeInfoDb        1.44.2     2025-08-20 [1] Bioconductor 3.21 (R 4.3.3)
   GenomeInfoDbData    1.2.14     2025-08-29 [1] Bioconductor
 P ggfun               0.2.0      2025-07-15 [?] CRAN (R 4.3.3)
   ggplot2           * 3.5.2      2025-04-09 [1] CRAN (R 4.3.3)
 P ggplotify           0.1.2      2023-08-09 [?] CRAN (R 4.3.2)
   ggtree            * 3.16.3     2025-07-16 [1] Bioconductor 3.21 (R 4.3.3)
   glue                1.8.0      2024-09-30 [1] CRAN (R 4.3.3)
 P gridGraphics        0.5-1      2020-12-13 [?] CRAN (R 4.3.2)
   gtable              0.3.6      2024-10-25 [1] CRAN (R 4.3.3)
   hms                 1.1.3      2023-03-21 [1] CRAN (R 4.3.3)
   htmltools           0.5.8.1    2024-04-04 [1] CRAN (R 4.3.3)
 P htmlwidgets         1.6.4      2023-12-06 [?] CRAN (R 4.3.2)
 P httpuv              1.6.16     2025-04-16 [?] CRAN (R 4.3.3)
   httr                1.4.7      2023-08-15 [1] CRAN (R 4.3.3)
   igraph              2.1.4      2025-01-23 [1] CRAN (R 4.3.3)
   IRanges             2.42.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   iterators           1.0.14     2022-02-05 [1] CRAN (R 4.3.3)
   jsonlite            2.0.0      2025-03-27 [1] CRAN (R 4.3.3)
   knitr               1.50       2025-03-16 [1] CRAN (R 4.3.3)
   labeling            0.4.3      2023-08-29 [1] CRAN (R 4.3.3)
 P later               1.4.4      2025-08-27 [?] CRAN (R 4.3.3)
   lattice             0.22-5     2023-10-24 [2] CRAN (R 4.3.3)
   lazyeval            0.2.2      2019-03-15 [1] CRAN (R 4.3.3)
   lifecycle           1.0.4      2023-11-07 [1] CRAN (R 4.3.3)
   lubridate         * 1.9.4      2024-12-08 [1] CRAN (R 4.3.3)
   magrittr            2.0.3      2022-03-30 [1] CRAN (R 4.3.3)
 P maps              * 3.4.3      2025-05-26 [?] CRAN (R 4.3.3)
   MASS                7.3-60.0.1 2024-01-13 [2] CRAN (R 4.3.3)
   Matrix              1.6-5      2024-01-11 [2] CRAN (R 4.3.3)
   memoise             2.0.1      2021-11-26 [1] CRAN (R 4.3.3)
   mgcv                1.9-1      2023-12-21 [2] CRAN (R 4.3.3)
   mime                0.13       2025-03-17 [1] CRAN (R 4.3.3)
 P miniUI              0.1.2      2025-04-17 [?] CRAN (R 4.3.3)
 P mnormt              2.1.1      2022-09-26 [?] CRAN (R 4.3.2)
   multtest            2.58.0     2023-10-24 [1] Bioconductor
   nlme                3.1-164    2023-11-27 [2] CRAN (R 4.3.3)
 P numDeriv            2016.8-1.1 2019-06-06 [?] CRAN (R 4.3.2)
 P optimParallel       1.0-2      2021-02-11 [?] CRAN (R 4.3.2)
 P pacman              0.5.1      2019-03-11 [?] CRAN (R 4.3.2)
   patchwork         * 1.3.2      2025-08-25 [1] CRAN (R 4.3.3)
   permute             0.9-8      2025-06-25 [1] CRAN (R 4.3.3)
 P phangorn            2.12.1     2024-09-17 [?] CRAN (R 4.3.2)
   phyloseq          * 1.52.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   phytools          * 2.4-4      2025-01-08 [1] CRAN (R 4.3.3)
   pillar              1.11.0     2025-07-04 [1] CRAN (R 4.3.3)
 P pkgbuild            1.4.8      2025-05-26 [?] CRAN (R 4.3.3)
   pkgconfig           2.0.3      2019-09-22 [1] CRAN (R 4.3.3)
 P pkgload             1.4.0      2024-06-28 [?] CRAN (R 4.3.2)
   plyr                1.8.9      2023-10-02 [1] CRAN (R 4.3.3)
 P profvis             0.4.0      2024-09-20 [?] CRAN (R 4.3.2)
 P promises            1.3.3      2025-05-29 [?] CRAN (R 4.3.3)
   purrr             * 1.1.0      2025-07-10 [1] CRAN (R 4.3.3)
 P quadprog            1.5-8      2019-11-20 [?] CRAN (R 4.3.2)
   R6                  2.6.1      2025-02-15 [1] CRAN (R 4.3.3)
   rappdirs            0.3.3      2021-01-31 [1] CRAN (R 4.3.3)
   RColorBrewer        1.1-3      2022-04-03 [1] CRAN (R 4.3.3)
   Rcpp                1.1.0      2025-07-02 [1] CRAN (R 4.3.3)
   readr             * 2.1.5      2024-01-10 [1] CRAN (R 4.3.3)
 P remotes             2.5.0      2024-03-17 [?] CRAN (R 4.3.2)
   reshape2            1.4.4      2020-04-09 [1] CRAN (R 4.3.3)
   rhdf5               2.46.1     2023-11-29 [1] Bioconductor 3.18 (R 4.3.3)
   rhdf5filters        1.14.1     2023-11-06 [1] Bioconductor
   Rhdf5lib            1.24.2     2024-02-07 [1] Bioconductor 3.18 (R 4.3.3)
   rlang               1.1.6      2025-04-11 [1] CRAN (R 4.3.3)
   rmarkdown           2.29       2024-11-04 [1] CRAN (R 4.3.3)
   rstudioapi          0.17.1     2024-10-22 [1] CRAN (R 4.3.3)
   S4Vectors           0.46.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   scales              1.4.0      2025-04-24 [1] CRAN (R 4.3.3)
 P scatterplot3d       0.3-44     2023-05-05 [?] CRAN (R 4.3.2)
 P sessioninfo         1.2.3      2025-02-05 [?] RSPM (R 4.3.3)
 P shiny               1.11.1     2025-07-03 [?] CRAN (R 4.3.3)
   stringi             1.8.7      2025-03-27 [1] CRAN (R 4.3.3)
   stringr           * 1.5.1      2023-11-14 [1] CRAN (R 4.3.3)
   survival            3.5-8      2024-02-14 [2] CRAN (R 4.3.3)
   tibble            * 3.3.0      2025-06-08 [1] CRAN (R 4.3.3)
   tidyr             * 1.3.1      2024-01-24 [1] CRAN (R 4.3.3)
   tidyselect          1.2.1      2024-03-11 [1] CRAN (R 4.3.3)
   tidytree          * 0.4.6      2023-12-12 [1] CRAN (R 4.3.3)
   tidyverse         * 2.0.0      2023-02-22 [1] CRAN (R 4.3.3)
   timechange          0.3.0      2024-01-18 [1] CRAN (R 4.3.3)
 P treeio              1.32.0     2025-04-15 [?] Bioconduc~
   tzdb                0.5.0      2025-03-15 [1] CRAN (R 4.3.3)
   UCSC.utils          1.4.0      2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
 P urlchecker          1.0.1      2021-11-30 [?] CRAN (R 4.3.2)
 P usethis             3.2.0      2025-08-28 [?] CRAN (R 4.3.3)
   vctrs               0.6.5      2023-12-01 [1] CRAN (R 4.3.3)
   vegan               2.7-1      2025-06-05 [1] CRAN (R 4.3.3)
   withr               3.0.2      2024-10-28 [1] CRAN (R 4.3.3)
   xfun                0.53       2025-08-19 [1] CRAN (R 4.3.3)
 P xtable              1.8-4      2019-04-21 [?] CRAN (R 4.3.2)
   XVector             0.48.0     2025-04-15 [1] Bioconductor 3.21 (R 4.3.3)
   yaml                2.3.10     2024-07-26 [1] CRAN (R 4.3.3)
 P yulab.utils         0.2.1      2025-08-19 [?] CRAN (R 4.3.3)

 [1] /local/workdir/arp277/Pendleton_2025_Absolute_Unifrac_Paper/renv/library/R-4.3/x86_64-pc-linux-gnu
 [2] /programs/R-4.3.3/lib64/R/library

 * ── Packages attached to the search path.
 P ── Loaded and on-disk path mismatch.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>